library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -259,7 +250,7 @@ public boolean handleMessage(@NonNull Message msg) {
             // TODO: handle media scanner timeouts
 
             final boolean isActive;
-            synchronized (downloads) {
+            synchronized (DownloadService.this) {
```

Comment:
so long as we ensure that the value objects returned by the repositories are copied and can't change